Modules/Sources/Yosemite/Tools/POS/POSCatalogSyncCoordinator.swift

```diff
@@ -446,6 +535,117 @@ public actor POSCatalogSyncCoordinator: POSCatalogSyncCoordinatorProtocol {
         // Record first sync date if needed
         recordFirstSyncIfNeeded(for: siteID)
     }
+
+    // MARK: - Analytics Helpers
+
+    private nonisolated func trackAnalytics(_ event: WooAnalyticsEvent) {
+        analytics?.track(event.statName.rawValue, properties: event.properties, error: event.error)
+    }
+
+    private nonisolated func getConnectionType() -> String {
+        guard let observer = connectivityObserver else { return "unknown" }
+        switch observer.currentStatus {
+        case .reachable(let connectionType):
+            switch connectionType {
+            case .ethernetOrWiFi:
+                return "wifi"
+            case .cellular:
+                return "cellular"
+            case .other:
+                return "unknown"
+            }
+        case .notReachable, .unknown:
+            return "unknown"
+        }
+    }
+
+    private func getStorageCounts(for siteID: Int64) async -> (products: Int, variations: Int) {
+        do {
+            return try await grdbManager.databaseConnection.read { db in
+                let productCount = try PersistedProduct.filter { $0.siteID == siteID }.fetchCount(db)
+                let variationCount = try PersistedProductVariation.filter { $0.siteID == siteID }.fetchCount(db)
+                return (productCount, variationCount)
+            }
+        } catch {
+            DDLogError("⛔️ POSCatalogSyncCoordinator: Failed to get storage counts: \(error)")
+            return (0, 0)
```

Comment:
Nit, just for readability
```suggestion
            return (products: 0, variations: 0)
```